contracts/FlyoverDiscovery.sol

```diff
@@ -0,0 +1,266 @@
+} from "@openzeppelin/contracts-upgradeable/access/extensions/AccessControlDefaultAdminRulesUpgradeable.sol";
+import {ICollateralManagement} from "./interfaces/ICollateralManagement.sol";
+import {IFlyoverDiscovery} from "./interfaces/IFlyoverDiscovery.sol";
+import {Flyover} from "./libraries/Flyover.sol";
+
+/// @title FlyoverDiscovery
+/// @notice Registry and discovery of Liquidity Providers (LPs) for Flyover
+/// @dev Keeps LP metadata and consults `ICollateralManagement` to decide listing and operational status
+contract FlyoverDiscovery is
+    AccessControlDefaultAdminRulesUpgradeable,
+    IFlyoverDiscovery
+{
+
+    // ------------------------------------------------------------
+    // FlyoverDiscovery State Variables
+    // ------------------------------------------------------------
+
+    mapping(uint => Flyover.LiquidityProvider) private _liquidityProviders;
+    ICollateralManagement private _collateralManagement;
+    uint public lastProviderId;
+
+    // ------------------------------------------------------------
+    // Basic Configuration State Variables
+    // ------------------------------------------------------------
+
+    uint private _minCollateral;
+    mapping(address => uint256) private _resignationBlockNum;
+
+    // ------------------------------------------------------------
+    // FlyoverDiscovery Public Functions and Modifiers
+    // ------------------------------------------------------------
+
+    /// @notice Initializes the contract with admin configuration
+    /// @dev Uses OZ upgradeable admin rules. Must be called only once
+    /// @param owner The Default Admin and initial owner address
+    /// @param initialDelay The initial admin delay for `AccessControlDefaultAdminRulesUpgradeable`
+    /// @param minCollateral The minimum collateral required per service side
+    /// @param collateralManagement The address of the `ICollateralManagement` contract
+    function initialize(
+        address owner,
+        uint48 initialDelay,
+        uint minCollateral,
+        address collateralManagement
+    ) external initializer {
+        __AccessControlDefaultAdminRules_init(initialDelay, owner);
+        _minCollateral = minCollateral;
+        _collateralManagement = ICollateralManagement(collateralManagement);
+    }
+
+    /// @notice Registers the caller as a Liquidity Provider
+    /// @dev Reverts if caller is not an EOA, already resigned, provides invalid data, invalid type, or lacks collateral
+    /// @param name Human-readable LP name
+    /// @param apiBaseUrl Base URL of the LP public API
+    /// @param status Initial status flag (enabled/disabled)
+    /// @param providerType The service type(s) the LP offers
+    /// @return id The newly assigned LP identifier
+    function register(
+        string calldata name,
+        string calldata apiBaseUrl,
+        bool status,
+        Flyover.ProviderType providerType
+    ) external payable returns (uint) {
```

Comment:
this function is not sending the collateral received to the CollateralManagement contract, that step is very important (rememer to give the COLLATERAL_ADDER role to this contract in the script to be able to do so)